Feat  update domoticz api version to 2023.2

DaKarlii opened the pull request:
Updated code because Plugin was not working with latest Domoticz releases due the changes on the API:
https://wiki.domoticz.com/Domoticz_API/JSON_URL's#Warning_Stable_2023.2_(and_newer)

Additionally the Support for SSL (Self-Signed) Connections to Domoticz were added.

Commit:
feat: Update Domoticz API and SSL Support

Commit:
feat: Add pyproject File to get rid off issue https://github.com/pypa/pip/issues/6334

Commit:
fix: Pass new value

Commit:
fix: Update Python Version

Commit:
feat: Update Version ID

Commit:
fix: pyproject.toml Delete (Keep Legacy Install Mode)

Commit:
Feat: Update GitIgnore

Commit:
feat: cleanUp Code

Commit:
feat: Update README.md for New Features

jneilliii commented:
Thanks for this PR, since I don't really use Domoticz personally I don't keep up with their development/release cycle. Do you by chance know if these changes will break previous versions of the API calls? It's hard to tell the exact changes because of the conversion of tabs to space indentation has made the whole init.py file be new.

DaKarlii commented:
Hi, 
Sry regarding the conversation from Tabs to Space - thats my VSC and not sure what triggers that.

To be honest i think it will break previous versions **IF** Domoticz is older then Version 2023.2 (Released July 21 2023).
https://github.com/domoticz/domoticz/blob/development/History.txt 
--> Line200 : "- Changed: API: Old RType calls have been replaced with normal calls"

With the current release 2025.1 the sld RType calls were completly removed.
--> Line60: "- Removed: API: Old RType calls have been replaced"

So until 2023.1: API RType
2023.2 until 2024.7: API Rtype + New/Normal
since 2025.1: API New/Normal only

I only use the current release and have no LAB for older ones.

Should i add a Warning to the Readme? 
("Attention Version Domoticz 2023.2 and higher is needed now" - some kind of that?)

DaKarlii commented:
Changes in Detail:

To Fix the API:
`json.htm?type=devices` was replaced by `type=command&param=getdevices`

And for SSL support:
`str_url = f"http://{plug_ip}` was replaced by `str_url = f"{plug_ip}`
Then the new option `ignoreSSL` was added to the definition `def on_api_command`.
and after that i changed the `if` loops to check if ignoreSSL is checked or not. (If Checked the Parameter `verify=False` will be added to the Response.)

Example:

```
if ignoreSSL:
  if username != "":
    web_response = requests.get(str_url, auth=(username, password), timeout=10, verify=False)
  else:
    web_response = requests.get(str_url, timeout=10, verify=False)
else:
  if username != "":
    web_response = requests.get(str_url, auth=(username, password), timeout=10)
  else:
    web_response = requests.get(str_url, timeout=10)
```

jneilliii commented:
Thanks, that's roughly what I gleamed from the code you provided. I think that API support for older versions isn't that critical, being it's 2 years old. I will however probably tweak the code a bit to automatically add the `http://` in front of existing installs to handle the conversion to the new format.

jneilliii committed:
add settings migration for new format

jneilliii commented:
I've pushed that change to your PR, if you don't mind setting your IP:PORT back to one without http(s):// in it and verify that when you install this updated version that it automatically adds the http:// in front then I think we're good to go with merge.

jneilliii committed:
fix settings migrations and python version

add ignoreSSL to settings migration
don't lock plugin into python 3.11 or newer.

jneilliii committed:
Update setup.py

bump major version due to potential breaking changes

DaKarlii commented:
I'm sorry to say but your changes broke the plugin now. 
It took some time to create a LAB to test the new changes (i didnt wanted to do it with my running systems)...

Your first change "add settins migration for new format" breaks the config page of the plugin. 
It is not possible to edit existing nor add new Plugs.


The Second change "fix settings migrations and python version" breaks the plugin completly. 
After PluginUpdate and service reload, the plugin stay in disabled mode and issue is `Incompatible to Octoprint or the underlying python environment`

DaKarlii commented:
~~Root Cause found - '__plugin_pythoncompat__ = ">=3.7,<4"' with this Parameter the Plugin Wont run.~~

~~I changed it Localy in my Lab back to original '__plugin_pythoncompat__ = ">=2.7,<4"'~~

~~With that the Plugin works well again. Including the setting migration!~~

Update: 
The Tabstops on Line124 were causing the issue. Replaced with Whitespaces and now test is OK!

DaKarlii committed:
Update __init__.py

Line 124:
Replace TABSTOPS with WHITESPACE

jneilliii commented:
Thanks, sorry about that with the tab vs space, my editor defaults to tabs. Thanks again for your contribution.
